### DISCUSSION TITLE
Is there a way to define a subgraph?

### DISCUSSION THREAD

**Discussion by r0ckarong** [monora/rgl#82] [Q&A] [answered]
I am looking for a way to generate graph data that is handled as a `subgraph` by dot as per the Graphviz definition:

https://graphviz.org/doc/info/lang.html#subgraphs-and-clusters

I would like to split some parts of my graph into subgraphs to control the rank and some styling options better than on the "full graph" level. I found the Subgraph method in the code documentation but this appears to be on the low level DOT implementation and not for the rgl interface itself (https://monora.github.io/rgl/RGL/DOT/Subgraph.html).

**Comment by monora** [comment]
Yes, this is more a Graphviz concept. RGL has a more algorithmic nature as already mentioned in #81. The [RGL::DOT](https://monora.github.io/rgl/RGL/DOT.html) module is only for convenience included for quick graph visualization.

So it depends which graph representation you have at hand. If you are on the Graphviz (DOT) level, you should eventually better work directly with the DOT language than with an RGL Graph.

**Reply by r0ckarong** [reply]
I'm actually generating nodes and edges from some other data and end up with an RGL data structure. So I only use the DOT implementation in RGL to write it out. I get pretty far but I'm still somewhat missing the subgraph for some additional freedom. 

![template-graph](https://user-images.githubusercontent.com/4920542/219710398-45a0125a-f7fc-46df-a20d-d892525268e6.png)

In this image, for example, we have different file types and their relationships. For some better control on how to position the nodes in the diagram I would create subgraphs from the different "colors" (nodes of similar type). I just wondered if this was already possible without diving too deep into DOT.

**Comment by monora** [comment]
I now understand your intention. I have never used the DOT subgraph feature together with RGL. So I guess you will have to experiment on your own. The only usage of DOT in RGL is in [Graph#to_ dot_graph](https://monora.github.io/rgl/RGL/Graph.html#to_dot_graph-instance_method). The `subgraph` feature is not used there. So only _flat_ graphs are drawn from RGL graphs.

Suppose you have a RGL graph which has your colored subgraphs as vertices and induced edges from the _flat_ graph. Then you could write a (rather complex) export to a dot graph with subgraphs...

If your only intention is to visualize your file structure with no need for RGLs algorithms I would generate DOT graphs directly.

**Comment by r0ckarong** [comment, marked as answer]
> if your only intention is to visualize your file structure with no need for RGLs algorithms I would generate DOT graphs directly.

For now, I only want to make sure that the diagram that is being generated is correct. I'm aiming towards using that code as a basis for something more "interactive" and then I will need graph traversal anyway.
